docs/source/tools/export.rst

```diff
@@ -0,0 +1,164 @@
+  script to understand the exported ledger state or history, and you can modify
+  this script for debugging or testing purposes.
+``testExport : Script ()``
+  A Daml script that will first invoke ``allocateParties`` and then ``export``.
+
+In most simple cases the generated Daml script will use the functions
+``submit`` or ``submitMulti`` to issue ledger commands that reproduce a
+transaction or ACS. In some cases the generated Daml script
+will fall back to the more general functions ``submitTree`` or
+``submitTreeMulti``.
+
+For example, the following generated code issues a create-and-exercise command
+that creates an instance of ``ContractA`` and exercises the choice ``ChoiceA``.
+The function ``submitTree`` returns a ``TransactionTree`` object that captures
+all contracts that are created in the transaction. The ``fromTree`` function is
+then used to extract the contract ids of the ``ContractB`` contracts that were
+created by ``ChoiceA``.
+
+.. code-block:: daml
+
+  tree <- submitTree alice_0 do
+    createAndExerciseCmd
+      Main.ContractA with
+        owner = alice_0
+      Main.ChoiceA
+  let contractB_1_1 = fromTree tree $
+        exercised @Main.ContractA "ChoiceA" $
+        created @Main.ContractB
+  let contractB_1_2 = fromTree tree $
+        exercised @Main.ContractA "ChoiceA" $
+        createdN @Main.ContractB 1
+
+.. TODO[AH] Add a full example project and example export.
+
+.. _export-ledger-offsets:
+
+Ledger Offsets
+**************
+
+By default ``daml ledger export`` will reproduce all transactions, as seen by
+the selected parties, from the beginning of the ledger history to the current
+end. The command-line flags ``--start`` and ``-end`` can be used to change this
+behavior. Both flags accept ledger offsets, either the special offsets
+``start`` and ``end``, or an arbitrary ledger offset.
+
+``--start``
+  Transactions up to and including the start offset will be reproduced as a
+  sequence of create commands that reproduce the ACS as of the start offset.
+  Later transactions will be reproduced as seen by the configured parties. In
+  particular, ``--start end`` will reproduce the current ACS but no transaction
+  history, ``--start start`` (the default) will reproduce the history of all
+  transactions as seen by the configured parties.
+``--end``
+  Export transactions up to and including this end offset.
+
+.. TODO[AH] Provide a reference or hints how to obtain arbitrary ledger offsets.
+
+.. _export-caveats:
+
+Caveats
+*******
+
+Unknown Contract Ids
+====================
+
+Daml ledger export does currently not handle references to unknown contract
+ids, but will just fail if any such references are encountered. This may occur
+if a contract was divulged to one of the configured parties, but the event that
+initially created that contract is not visible to any of the configured
+parties. This may also occur if a contract was archived before the configured
```

Comment:
Maybe worth clarifying that this is only an issue if the contract was archived _and_ you also retained a reference to it. Archived contracts in general are fine.